### PR TITLE
Test reachability service

### PR DESCRIPTION
**Pull request by raphrel**

### CODE REVIEW

File: Sources/Services/ReachabilityService.swift [hunk 1]
```diff
@@ -8,32 +8,40 @@
...
+    private var timer: TimerProtocol?
     private var tiNetworkTesting: TimeInterval
-    private var networkTask: URLSessionTask?
-    private var session: URLSession {
-        let config = URLSessionConfiguration.ephemeral
-        config.timeoutIntervalForRequest = timeoutURLSession
-        return URLSession(configuration: config)
-    }
+    private var networkTask: URLSessionDataTaskProtocol?
+    private var dataTaskFactory: URLSessionDataTaskFactoryProtocol
```
[the remark this answers is not in the document]
Comment (raphrel) [reply Sources/Services/ReachabilityService.swift]: yes

File: Sources/State/FailedState.swift [hunk 2]
```diff
@@ -18,18 +18,19 @@ public final class FailedState: PlayerState {
     // MARK: - Private vars
 
     private let urlToReload: URL
-    private let reachability: ReachabilityService
+    private var reachability: ReachabilityServiceProtocol
```
[the remark this answers is not in the document]
Comment (raphrel) [reply Sources/State/FailedState.swift]: nope car on set le callback un peu plus bas.

File: Tests/Services/ReachabilityServiceTests.swift [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+                    
+                    // ACT
+                    self.tested.start()
+                    let maxIterationAvailable = self.config.networkIteration
+                    var iteration = 0
+                    
+                    while iteration < maxIterationAvailable {
+                        self.mockTimerFactory.lastCompletion?()
+                        iteration += 1
+                    }
```
[the remark this answers is not in the document]
Comment (raphrel) [reply Tests/Services/ReachabilityServiceTests.swift]: yes!

File: Tests/Services/ReachabilityServiceTests.swift [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+                    self.mockTimerFactory.lastCompletion?()
+                }()
+                
+                // ASSERT
+                expect(self.dataTaskFactory.dataTask.cancel_CallCount).to(equal(1))
+            }
+        }
+        
+        describe("start service") {
+
```
[the remark this answers is not in the document]
Comment (raphrel) [reply Tests/Services/ReachabilityServiceTests.swift]: yup

File: Sources/State/FailedState.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (JiceDessaint) [reply Sources/State/FailedState.swift]: ok

**Pull request merged**